python/pyarrow/dataset.py

```diff
@@ -694,22 +697,23 @@ def _ensure_write_partitioning(scheme):
     return scheme
 
 
-def write_dataset(data, base_dir, format=None, partitioning=None, schema=None,
-                  filesystem=None, use_threads=True):
+def write_dataset(data, base_dir, basename_template, format=None,
```

Comment:
Should we provide a default template here? 
Can eg the format object have a property with the default name to use? (or get the extension from there and use that in a default?)

Reply:
That sounds fine but I'd like to defer that to a follow up